### ISSUE TITLE
robot reduce should remove redundant type axioms

### ISSUE THREAD

**Issue by dosumis** [ontodev/robot#989]
ROBOT REDUCE does not remove classification (Typing) of DL1_aDPN in this case:

```
Declaration(Class(<http://scratch#cell>))
Declaration(Class(<http://scratch#neuron>))
Declaration(NamedIndividual(<http://scratch#DL1_adPN>))
SubClassOf(<http://scratch#neuron> <http://scratch#cell>)
ClassAssertion(<http://scratch#cell> <http://scratch#DL1_adPN>)
ClassAssertion(<http://scratch#neuron> <http://scratch#DL1_adPN>)
```

![image](https://user-images.githubusercontent.com/112839/163988687-23ec2e34-6c5e-421f-81ac-3b93d4b066e8.png)

![image](https://user-images.githubusercontent.com/112839/163988480-e4cf9be5-cd12-4151-af4f-8e9e443ad0a2.png)

It should.  

Please can this be added?

**Comment by dosumis** [comment]
CC @hkir-dev is available to implement this if we agree the change should be made.

**Comment by matentzn** [comment]
I support this change, it's an extension of reduce which is totally in the spirit of it. Probably we need a parameter to retain backwards compatibility.. what about --axioms with OWLAPI axiom types as parameters? This keeps us flexible moving forward.

**Comment by beckyjackson** [comment]
I also support this. Currently the docs state that `reduce` only removes subClassOf axioms. If we're going to go the route that @matentzn suggested, then I think it makes sense to implement everything at once and allow the following for the `--axioms` option: subclass, type, subProperty (differentiate between data/object?)... there's also domains & values to consider (would these be used?). That's a lot of work to just get it working for individuals, though.

**Comment by jamesaoverton** [comment]
I agree that we should add this with a new option. We can start small or with low-hanging fruit, but let's continue this discussion and be very clear about what we want to do.

**Comment by dosumis** [comment]
I think an MVP implementation would extend only to support redundancy stripping of type assertions from individuals with argument  `--axioms subClass Type`.  Let's leave subProperty to a later iteration.
